### PR TITLE
Advisor Remediation Issue : ADVISOR-2388 | DO NOT MERGE UNTILL 2472 IS MERGED

### PR DESCRIPTION
**Pull request by adonispuente**
The new redux bulk Select handeling is included in this PR, and therefor cannot be merged until changes for ADVISOR-2472 are approved.

This PR is in regard to ticket ADVISOR-2388, having the option to go through remediation process with systems that are not remediable (manual remediations).
This see changes ,

Run the app against prod DB (only prod has manual remediations_ with CHROME_ENV=prod-stable npm run start:proxy:beta
navigate to recommendations, and then click pathways
Click the Update Service Configuration pathway, this is the only known pathway with a manual remediation (if you find another on your account that works too)
if you click any of the systems, the button will stay disabled as they cannot be remediated automatically.
However, if you were to find a pathway that had both manual and automatic remediable systems, the button would be clickable as long as at least one of the selected systems can be remediated automatically.

### CODE REVIEW

**Review by mkholjuraev — COMMENTED** [review]
It looks clean and easy to read PR. I enjoyed reviewing and left some suggestions. 

While I was testing locally, I could see that remdiation button stayed disabled when systems of Update Service Configuration selected. I was not able find a pathway which has systems which are both remediatable and not. Because, I do not know what proves that a system is not remediatable.

If you can provide me, I can test the second case as well.

File: src/PresentationalComponents/Inventory/Hooks/useSelectionManager/reducer.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  );
+  return cleanEmpty({
+    ...state,
+    [group]: items.length === 0 ? undefined : items,
+  });
+};
+
+const toggle = (state, action) => {
+  const group = selectionGroup(action);
+  // eslint-disable-next-line prettier/prettier
```
[the remark this answers is not in the document]
Comment (adonispuente) [reply src/PresentationalComponents/Inventory/Hooks/useSelectionManager/reducer.js:53]: Seems you're right, with this it stopped complaining

File: src/PresentationalComponents/Inventory/Hooks/useBulkSelect/helpers.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    return `${itemsTotal} selected`;
+  }
+};
+
+export const checkboxState = (selectedItemsTotal, itemsTotal) => {
+  if (selectedItemsTotal === 0) {
+    return false;
+  } else if (selectedItemsTotal === itemsTotal) {
+    return true;
+  } else {
```
[the remark this answers is not in the document]
Comment (adonispuente) [reply src/PresentationalComponents/Inventory/Hooks/useBulkSelect/helpers.js:16]: Oh youre right, for some reason I felt very strongly that I needed to keep this

File: src/PresentationalComponents/Inventory/Inventory.js [hunk 3]
```diff
@@ -45,19 +46,128 @@ const Inventory = ({
...
-  const onSelectRows = (id, selected) =>
-    dispatch({ type: 'SELECT_ENTITY', payload: { id, selected } });
   const addNotification = (data) => dispatch(notification(data));
   const [disableRuleModalOpen, setDisableRuleModalOpen] = useState(false);
-  const [bulkSelect, setBulkSelect] = useState();
+  const [curPageIds, setCurPageIds] = useState([]);
+  const [pathwayRulesList, setPathwayRulesList] = useState();
+  const [pathwayReportList, setPathwayReportList] = useState();
+
+  const [pathwayDetails, setPathwayDetails] = useState(false);
```
[the remark this answers is not in the document]
Comment (adonispuente) [reply src/PresentationalComponents/Inventory/Inventory.js:63]: I agree, this is definitely easier to read

**Review by mkholjuraev — DISMISSED** [review]
LGTM! thanks @adonispuente.

**Comment by adonispuente** [comment]
I accidentally left a linting error when I was fixing a merge conflict and had to repush

**Review by mkholjuraev — APPROVED** [review]

**Pull request merged**